vector_index.go

```diff
@@ -48,8 +48,9 @@ type VectorReader interface {
 }
 
 type VectorIndexReader interface {
-	VectorReader(ctx context.Context, vector []float32, field string, k int64, searchParams json.RawMessage) (
-		VectorReader, error)
+	VectorReader(ctx context.Context, vector []float32, field string, k int64,
+		searchParams json.RawMessage, filterIDs []IndexInternalID,
+		requireFiltering bool) (VectorReader, error)
```

Comment:
Let's add a new method for VectorReadWithFilter(..) instead.

Also, I don't quite understand the need for `requireFiltering` here.

Reply:
Like mentioned in my earlier comment, requireFiltering is to distinguish between filtered and unfiltered kNN.

Reply:
Now, that i've added a separate reader for filtered kNN, it's no longer required and hence, i've removed it in a recent commit.